### DISCUSSION TITLE
SOPS v3.8.0-rc.1

### DISCUSSION THREAD

**Discussion by hiddeco** [getsops/sops#1270] [Announcements]
After a long period of silence, we are happy to share the announcement of [release candidate `v3.8.0-rc.1`](https://github.com/getsops/sops/releases/tag/v3.8.0-rc.1)! :tada: 

This announcement serves as the primary place to discuss this release, and report potential issues with this release candidate **compared to `v3.7.3`**. When you are running into an issue which can be reproduced with `v3.7.3`, please open [a new issue instead](https://github.com/getsops/sops/issues/new).

For the full changelog, refer to the [`CHANGELOG.rst` file](https://github.com/getsops/sops/blob/v3.8.0-rc.1/CHANGELOG.rst), or the link to the release page above.

**Comment by mxm-tr** [comment]
Hi, when running a decryption using PGP keys with the v3.8.0-rc.1 version, we get a new error message on stdout.

We expect the decryption to try all PGP keys referenced in the .sops.yaml file, until it succeeds. In the `v3.8.0-rc.1`, each decryption attempt generates an error message on stdout:

Using `v3.7.3`: No output on stdout, resulting file OK

```text
[user@laptop deployment-values ]$ sops --version
sops 3.7.3
[user@laptop deployment-values ]$ sops --config <(echo '') -d --ignore-mac --output ./out.yml ./inventory.secrets.yaml 
```

Using `v3.8.0-rc.1`: PGP decryption errors on stdout, resulting file OK

```text
[user@laptop deployment-values ]$ sops-3.8 --config <(echo '') -d --ignore-mac --output ./out.yml ./inventory.secrets.yaml  
[PGP]    ERRO[0000] Decryption failed                             fingerprint=keyid1
[PGP]    ERRO[0000] Decryption failed                             fingerprint=keyid2
[PGP]    ERRO[0000] Decryption failed                             fingerprint=keyid3
[PGP]    ERRO[0000] Decryption failed                             fingerprint=keyid4
[user@laptop deployment-values ]$ sops-3.8 --version
sops 3.8.0-rc.1 (latest)

```

Let me know if you need any further information!

**Reply by hiddeco** [reply]
Thanks for the feedback, without looking into specific details this appears to be a regression. I will take a deep dive tonight and see if I can either explain the difference, or issue a fix.

**Reply by felixfontein** [reply]
One important clarification: the output is made to stderr, **not** to stdout.

**Reply by felixfontein** [reply]
I think the problem is this line: https://github.com/getsops/sops/blob/main/pgp/keysource.go#L382

Removing it probably solves the problem. If decrypting finally fails, sops prints a summary of all decryption attempts anyway.

**Reply by felixfontein** [reply]
The logging in this case was introduced in https://github.com/hiddeco/sops/commit/c6236ada5645b6c08e0d87dbb946fcc6d1cfc1a1#diff-7b16bb58ed45315df8ef61219085e34408638b7fbd87404e036fb91971014b82R334

**Reply by hiddeco** [reply]
But https://github.com/getsops/sops/blob/v3.7.3/pgp/keysource.go#L233 :thinking:

**Reply by felixfontein** [reply]
That's an *info* logging, not an *error* logging.

**Reply by hiddeco** [reply]
_OOH_, this then makes this related to https://github.com/getsops/sops/pull/1146 and is a regression.

**Reply by mxm-tr** [reply]
Thanks for looking into it 🙏

**Reply by felixfontein** [reply]
Fixed by #1281; I just tested with a binary built with that PR merged, and it no longer shows the warnings.
